test/explorer/series/struct_test.exs

```diff
@@ -0,0 +1,139 @@
+             ]
+    end
+
+    test "allows structs structs with special float values" do
+      series = Series.from_list([%{a: :nan, b: :infinity, c: :neg_infinity}])
+
+      assert series.dtype == {:struct, %{"a" => {:f, 64}, "b" => {:f, 64}, "c" => {:f, 64}}}
+      assert series[0] == %{"a" => :nan, "b" => :infinity, "c" => :neg_infinity}
+      assert Series.to_list(series) == [%{"a" => :nan, "b" => :infinity, "c" => :neg_infinity}]
+    end
+
+    test "allows structs mixing integers and floats" do
+      series = Series.from_list([%{a: 1, b: 2.4}, %{a: 1.5, b: 2}])
+
+      assert series.dtype == {:struct, %{"a" => {:f, 64}, "b" => {:f, 64}}}
+      assert Series.to_list(series) == [%{"a" => 1.0, "b" => 2.4}, %{"a" => 1.5, "b" => 2.0}]
+    end
+
+    test "errors when structs have mismatched types" do
+      assert_raise ArgumentError,
+                   "the value %{a: \"a\"} does not match the inferred series dtype {:struct, %{\"a\" => :integer}}",
+                   fn -> Series.from_list([%{a: 1}, %{a: "a"}]) end
+
+      assert_raise ArgumentError,
+                   "the value %{b: 1} does not match the inferred series dtype {:struct, %{\"a\" => :integer}}",
+                   fn -> Series.from_list([%{a: 1}, %{b: 1}]) end
+    end
+  end
+
+  describe "cast/2" do
+    test "struct with integers to struct with floats" do
+      s = Series.from_list([%{a: 1}, %{a: 2}])
+      s1 = Series.cast(s, {:struct, %{"a" => {:f, 64}}})
+
+      assert Series.to_list(s1) == [%{"a" => 1.0}, %{"a" => 2.0}]
+      assert Series.dtype(s1) == {:struct, %{"a" => {:f, 64}}}
+    end
+
+    test "nested structs with integers to nested structs with floats" do
+      s = Series.from_list([%{a: %{b: 1}}, %{a: %{b: 2}}])
+      s1 = Series.cast(s, {:struct, %{"a" => {:struct, %{"b" => {:f, 64}}}}})
+
+      assert Series.to_list(s1) == [%{"a" => %{"b" => 1.0}}, %{"a" => %{"b" => 2.0}}]
+      assert Series.dtype(s1) == {:struct, %{"a" => {:struct, %{"b" => {:f, 64}}}}}
+    end
+
+    test "structs with integers to structs with datetimes" do
+      s =
+        Series.from_list([
+          %{a: 1},
+          %{a: 2},
+          %{a: 3},
+          %{a: 1_649_883_642 * 1_000 * 1_000}
+        ])
+
+      s1 = Series.cast(s, {:struct, %{"a" => {:datetime, :microsecond}}})
+
+      assert Series.to_list(s1) == [
+               %{"a" => ~N[1970-01-01 00:00:00.000001]},
+               %{"a" => ~N[1970-01-01 00:00:00.000002]},
+               %{"a" => ~N[1970-01-01 00:00:00.000003]},
+               %{"a" => ~N[2022-04-13 21:00:42.000000]}
+             ]
+
+      assert Series.dtype(s1) == {:struct, %{"a" => {:datetime, :microsecond}}}
+    end
+  end
+
+  describe "inspect/1" do
+    test "struct with integer values" do
+      s = Series.from_list([%{a: 1}, %{a: 2}])
+
+      assert inspect(s) ==
+               """
+               #Explorer.Series<
+                 Polars[2]
+                 struct[1] [%{"a" => 1}, %{"a" => 2}]
```

Comment:
I followed the Polars approach here and only displayed the number of keys in the struct series. This made sense to me, as it would get out of hand pretty quick with large structs and nested values. 

Let me know if you disagree and believe this should be more descriptive.